__tests__/app.test.js

```diff
@@ -92,3 +94,240 @@ describe("/api", () => {
+    const voteChange = { inc_votes: 5 };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(voteChange)
+      .expect(200);
+  });
+  test("should return updated version of article with votes incremented by specified amount", () => {
+    const voteChange = { inc_votes: 5 };
+    return request(app)
+      .patch("/api/articles/3")
+      .send(voteChange)
+      .then(({ body }) => {
+        const { article } = body;
+        expect(article.article_id).toBe(3);
+        expect(article.title).toBe(
+          "Eight pug gifs that remind me of mitch"
+        );
+        expect(article.topic).toBe("mitch");
+        expect(article.author).toBe("icellusedkars");
+        expect(article.body).toBe("some gifs");
+        expect(article.created_at).toBe("2020-11-03T09:12:00.000Z");
+        expect(article.votes).toBe(5);
+        expect(article.article_img_url).toBe(
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+        );
+      });
+  });
+  test("should return updated version of article with votes decremented by specified amount", () => {
+    const voteChange = { inc_votes: -100 };
+    return request(app)
+      .patch("/api/articles/3")
+      .send(voteChange)
+      .then(({ body }) => {
+        console.log(body);
+        const { article } = body;
+        expect(article.article_id).toBe(3);
+        expect(article.title).toBe(
+          "Eight pug gifs that remind me of mitch"
+        );
+        expect(article.topic).toBe("mitch");
+        expect(article.author).toBe("icellusedkars");
+        expect(article.body).toBe("some gifs");
+        expect(article.created_at).toBe("2020-11-03T09:12:00.000Z");
+        expect(article.votes).toBe(-100);
+        expect(article.article_img_url).toBe(
+          "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+        );
+      });
+  });
+  test("should return 404 error when valid article_id entered but does not exist", () => {
+    const voteChange = { inc_votes: 54 };
+    return request(app)
+      .patch("/api/articles/2000000")
+      .send(voteChange)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("article does not exist");
+      });
+  });
+  test("should return 400 error when invalid article_id entered", () => {
+    const voteChange = { inc_votes: 4 };
+    return request(app)
+      .patch("/api/articles/notanumber")
+      .send(voteChange)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
+  });
+  test("should return 400 error when provided vote change as an invalid datatype", () => {
+    const voteChange = { inc_votes: "STOP THE COUNT!!" };
+    return request(app)
+      .patch("/api/articles/1")
+      .send(voteChange)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("bad request");
+      });
```

Comment:
as well if the vote count isn't the right data type, what about if the inc_votes property was missing altogether? It's worth adding one more error test for this one